### DISCUSSION TITLE
How to add a confirmation popup before the dialog is closed?

### DISCUSSION THREAD

**Discussion by devalphabyte** [serenity-is/Serenity#5535] [Help] [answered]
Hi, I am trying to add a confirmation message using Q.Confirm("text here", () => { //close dialog if positive response });
how can I go about this?

**Comment by VictorTomaili** [comment]
Check dialog base class methods for override what required to you.

**Comment by luongnv6** [comment, marked as answer]
see here
https://github.com/serenity-is/Serenity/discussions/5407

**Comment by devalphabyte** [comment]
Thanks @luongnv6 
Link here: https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Core/Modules/Northwind/Customer/CustomerDialog.ts
